src/lua/tweaks.c

```diff
@@ -0,0 +1,135 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2023, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#include "lua/tweaks.h"
+
+#include <lua.h>
+#include <lauxlib.h>
+#include <stdbool.h>
+#include <stddef.h>
+
+#include "core/tweaks.h"
+#include "diag.h"
+#include "lua/utils.h"
+#include "trivia/util.h"
+
+/**
+ * Pushes a tweak value to Lua stack and returns the number of values pushed.
+ */
+static int
+luaT_push_tweak_value(struct lua_State *L, const struct tweak_value *v)
+{
+	switch (v->type) {
+	case TWEAK_VALUE_BOOL:
+		lua_pushboolean(L, v->bval);
+		return 1;
+	case TWEAK_VALUE_INT:
+		lua_pushinteger(L, v->ival);
+		return 1;
+	case TWEAK_VALUE_STR:
+		lua_pushstring(L, v->sval);
+		return 1;
+	default:
+		unreachable();
+	}
+}
+
+/**
+ * Looks up a tweak value by name (2nd arg) and pushes it to Lua stack.
+ * Returns nothing if there's no such tweak.
+ */
+static int
+luaT_tweaks_index(struct lua_State *L)
+{
+	const char *name = luaL_checkstring(L, 2);
+	struct tweak *t = tweak_find(name);
+	if (t == NULL)
+		return 0;
+	struct tweak_value v;
+	tweak_get(t, &v);
+	return luaT_push_tweak_value(L, &v);
+}
+
+/**
+ * Updates a tweak value given its name (2nd arg) and the new value (3rd arg).
+ * Raises an error if there's no such tweak or the value is invalid.
+ */
+static int
+luaT_tweaks_newindex(struct lua_State *L)
+{
+	const char *name = luaL_checkstring(L, 2);
+	struct tweak *t = tweak_find(name);
+	if (t == NULL) {
+		diag_set(IllegalParams, "No such option");
+		return luaT_error(L);
+	}
+	struct tweak_value v;
+	switch (lua_type(L, 3)) {
+	case LUA_TBOOLEAN:
+		v.type = TWEAK_VALUE_BOOL;
+		v.bval = lua_toboolean(L, 3);
+		break;
+	case LUA_TNUMBER:
+		v.type = TWEAK_VALUE_INT;
+		v.ival = lua_tointeger(L, 3);
+		if (lua_tonumber(L, 3) != v.ival)
+			goto invalid_value;
+		break;
```

Comment:
Nit: I guess I can pass a value above 2^32 and I'll not receive an error.

Reply:
You will receive an error if you pass a value that doesn't fit into `int`. I added a test case for this.

<details><summary>diff</summary>

```diff
diff --git a/test/unit/lua_tweaks.c b/test/unit/lua_tweaks.c
index bad4f993fb53..28d0caed656e 100644
--- a/test/unit/lua_tweaks.c
+++ b/test/unit/lua_tweaks.c
@@ -63,7 +63,7 @@ test_index(void)
 static void
 test_newindex(void)
 {
-	plan(6);
+	plan(8);
 	header();
 	is(luaT_dostring(L, "tweaks.no_such_var = 1"), -1, "unknown option");
 	ok(!diag_is_empty(diag_get()) &&
@@ -81,6 +81,12 @@ test_newindex(void)
 	   strcmp(diag_last_error(diag_get())->errmsg,
 		  "Value must be boolean, integer, or string") == 0,
 	   "check error");
+	is(luaT_dostring(L, "tweaks.bool_var = 8589934592"), -1,
+	   "invalid value - big int");
+	ok(!diag_is_empty(diag_get()) &&
+	   strcmp(diag_last_error(diag_get())->errmsg,
+		  "Value must be boolean, integer, or string") == 0,
+	   "check error");
 	footer();
 	check_plan();
 }
```

</details>

Reply:
`tweaks.bool_var` is a typo?

Reply:
No, it's not a typo. It doesn't make any difference which variable I use here, because I check the error message, and this error message is raised by `lua/tweaks.c`. If the value passed this check, the error would be different ("Invalid value, expected boolean"). I use `bool_var` for all other checks in this function.